src/components/AssetTypesComponent.jsx

```diff
@@ -1,98 +1,161 @@
+      (this.state.activePage * this.state.limit));
+    this.setState({
+      currentAssetTypes
+    });
+  };
 
-  loadRoles = () => {
-    if (this.emptyAssetTypeCheck()) {
-      return <Table.Row><Table.Cell colSpan="3">No Data found</Table.Cell></Table.Row>
-    } else {
-      const assetTypes = this.props.assetTypes.map((assetType, index) => {
-        return <TableRowComponent
-          key={index}
-          data={assetType}
-          headings={['category', 'sub_category', 'asset_type']}
-        />
-      });
-      return assetTypes;
-    }
+  handlePaginationChange = (e, { activePage }) => {
+    this.setState({
+      activePage,
+      offset: this.getOffset(activePage, this.state.limit)
+    }, () => this.setCurrentAssetTypes());
   }
 
+  getTotalPages = () => Math.ceil(this.props.assetTypes.length / this.state.limit)
+
+  emptyAssetTypeCheck = () => (this.props.assetTypes.length === 0)
+
+  // loadRoles = () => {
+  //   let assetTypes = null;
+  //   if (this.emptyAssetTypeCheck()) {
+  //     assetTypes = <Table.Row><Table.Cell colSpan="2">No Data found</Table.Cell></Table.Row>;
+  //   } else {
+  //     assetTypes = this.state.currentAssetTypes.map(assetType => (
+  //       <TableRowComponent
+  //         key={assetType.id}
+  //         data={assetType}
+  //         headings={['id', 'asset_type', 'asset_sub_category']}
+  //         action
+  //       />
+  //     ));
+  //   }
+  //   return assetTypes;
+  // }
+
   render() {
+    if (this.props.isLoading) {
+      return (
+        <SideMenuComponent>
+          <LoaderComponent size="large" dimmerStyle={{ height: '100vh' }} />
+        </SideMenuComponent>
+      );
+    } else if (!this.props.isLoading && _.isEmpty(this.props.assetTypes)) {
+      return (
+        <SideMenuComponent>
+          <Container>
+            <h1>
+              Unable to load Asset Types, refresh your browser or log out and in again.
+            </h1>
+          </Container>
+        </SideMenuComponent>
+      );
+    }
     return (
       <SideMenuComponent>
-        <Container>
-          <Header className='landing-heading' content='Asset Types' />
-          <Table celled>
-            <Table.Header>
-              <Table.Row>
-                <Table.HeaderCell>Category</Table.HeaderCell>
-                <Table.HeaderCell>Sub-category</Table.HeaderCell>
-                <Table.HeaderCell>Type</Table.HeaderCell>
-              </Table.Row>
-            </Table.Header>
-
-            <Table.Body>
-              {
-                this.loadRoles()
-              }
-            </Table.Body>
-
-            <Table.Footer>
-              <Table.Row>
-                <Table.HeaderCell colSpan='3'>
-                  {
-                    (this.emptyAssetTypeCheck()) ? '' :
+        {(this.emptyAssetTypeCheck()) ?
+          <LoaderComponent
+            size="large"
+            loadingText="loading..."
+          /> :
+          <Container>
```

Comment:
The ternary is extra long making it hard to read the code, especially the `<Container>...</Container>` part.  Therefore, I'm going to ask you again to make some changes to the render function. I.e.
```
render() {
  if (this.props.isLoading) {
    return (...);
  }

  if (!this.props.isLoading && _.isEmpty(this.props.assetTypes)) {
    return (...);
  }

  if (this.emptyAssetTypeCheck()) {
    return (
      <SideMenuComponent>
        <LoaderComponent
          size="large"
          loadingText="loading..."
        />
      </SideMenuComponent>
    );
  }

  return (
    <SideMenuComponent>
      <Container>
        ...
        ...
      </Container>
    </SideMenuComponent>
  );
}
```

**Note** that use of `else if` is unnecessary due to the return statements